src/tpi/primitive.rs

```diff
@@ -135,29 +135,29 @@ pub enum PrimitiveKind {
     HRESULT,
 }
 
-/// Pointer kinds.
+/// Pointer mode of primitive types.
+///
+/// This is partially overlapping with [`PointerKind`](enum.PointerKind.html) for regular pointer
+/// type definitions. While `PointerKind` can specify many more pointer types, including relative
+/// pointers, `Indirection` also contains a 128-bit variant.
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub enum Indirection {
-    ///
+    /// This type is not a pointer.
     None,
```

Comment:
Looking at this again, should we drop `Indirection::None` and change `PrimitiveType` to `pub indirection: Option<Indirection>`? In retrospect it's kinda weird to have a `Pointer kind` enumeration with `This type is not a pointer`, and we're breaking users of this `enum` anyway…

Reply:
I was thinking about this briefly, too. Let's do it 👍